### ISSUE TITLE
Suggestions for the development mode

### ISSUE THREAD

**Issue by mirukuz** [awohletz/electron-prisma-trpc-example#2]
I noticed there's dev mode in previous repo which can watch the changes on the client and hot reload the app.
maybe we can add these two new script in the `package.json` file?

```
"dev:client": "vite",
"dev:server": "tsc --build && node copy-files.js && cross-env NODE_ENV=development electron ."
```

and in `main.ts`

```ts
const isDev = process.env.NODE_ENV === "development";
const port = 5173; // Hardcoded; needs to match the port Vite runs the dev server on
const selfHost = `http://localhost:${port}`;
...
if (isDev) {
    win.loadURL(selfHost);
  } else {
    win.loadFile(path.join(__dirname, '..', 'index.html'));
  }
```

**Comment by awohletz** [comment]
I'd prefer to keep this repo as minimal as possible because:
1. It makes it easier to debug and keep up to date. There are no other interfering libraries that we have to worry about.
2. It makes it easy for devs to pick and choose what they want from it. I've noticed people have widely varying Electron setups -- Vanilla JS, React, Vue, Webpack, Vite, etc. This minimal demonstration keeps it relatively simple to bring in tRPC, Prisma, or both into heterogenous setups without the baggage of other opinions.

It's a good suggestion though. I've been thinking of forking a few repos from this one for common setups like hot-reloading React. Do you use React?

**Comment by mirukuz** [comment]
Yes, I use react, forking is a good idea! also got to know Vite from your code, looks much lightweight than react-scripts 👍
